assets/blocks/button/index.js

```diff
@@ -38,6 +38,18 @@ export const createButtonBlockType = ( { settings, ...options } ) => {
 		tagName: 'a',
 		...options,
 	};
+
+	const styles = settings.styles
+		? settings.styles
+		: [ { ...BlockStyles.Fill, isDefault: true }, BlockStyles.Outline ];
+
+	const defaultStyle = find( styles, 'isDefault' )?.name;
+
+	// eslint-disable-next-line @wordpress/no-unused-vars-before-return -- We don't wanna recreate the component every edit render.
+	const EditButtonBlockWithBlockStyle = withDefaultBlockStyle( defaultStyle )(
```

Comment:
It's more to avoid recreating the component (not the instance) every time the `edit` is called, which could cause some issues.

Reply:
What kind of issues can this cause? A bug or performance? I think we should avoid the extra complexity unless there is an indication that there is a degrade in performance.

Reply:
> What kind of issues can this cause? A bug or performance?

It's more susceptible to bugs than a performance issue.

Considering we have to enhance the component w/ the HOC and also pass the `options` we're creating here, we couldn't do something like:

```js
edit: withDefaultBlockStyle( defaultStyle )(EditButtonBlock)
```

And if we enhance the component inside the `edit` render scope (to prevent the eslint error), it would re-create the component function every edit render, which could cause many issues in React. In this case, every render we would create a component instance of a different component.

Thinking a little more, we could add a new argument to the `withDefaultBlockStyle` to send the `options` to add to the `props`, but it wouldn't be nice because it's not the purpose of this HOC, and it wouldn't be much explicit.